pages/scorepage/[weekday].js

```diff
@@ -14,23 +15,41 @@ export default function ScorePage({
   const router = useRouter();
 
   function handleSaveAndStartPage() {
+    const mergedArrayAllFoodAndSelectedFood = mergeArrayAllFoodAndSelectedFood(
+      foodCategories,
+      selectedFoodCategories
+    );
+
     const newObjectForWeeklyCollection = {
       weekday: currentWeekDay,
-      data: selectedFoodCategories,
+      data: mergedArrayAllFoodAndSelectedFood,
     };
 
-    handleDailyQuizzesResultCollection([
-      ...dailyQuizzesResultCollection,
-      newObjectForWeeklyCollection,
-    ]);
-    router.push("/");
+    let quizForWeekdayAlreadyExists = false;
```

Comment:
i am not totally sure, but would it be possible to use useState here instead of a let? to stay with const ?

like
 const [quizForWeekdayAlreadyExists,  setquizForWeekdayAlreadyExists] = useState(false)

and 
setQuizForWeekdayAlreadyExists(true) 
in the condition

Reply:
good point! I changed it to state, that's better. Thanks alot! :-)